syntax = "proto3";

package NAlice;

option go_package = "a.yandex-team.ru/alice/megamind/protos/blackbox";
option java_package = "ru.yandex.alice.megamind.protos.blackbox";
option java_outer_classname = "BlackboxUserInfo";

import "mapreduce/yt/interface/protos/extension.proto";

//TODO: reduce copy-paste with BASS .sc structure
message TBlackBoxUserInfo {
    string Uid = 1 [json_name = "uid", (NYT.column_name) = "uid"];
    string Email = 2 [json_name = "email", (NYT.column_name) = "email"];
    string FirstName = 3 [json_name = "firstName", (NYT.column_name) = "first_name"];
    string LastName = 4 [json_name = "lastName", (NYT.column_name) = "last_name"];
    string Phone = 5 [json_name = "phone", (NYT.column_name) = "phone"];
    bool HasYandexPlus = 6 [json_name = "hasYandexPlus", (NYT.column_name) = "has_yandex_plus"];
    bool IsStaff = 7 [json_name = "isStaff", (NYT.column_name) = "is_staff"];
    bool IsBetaTester = 8 [json_name = "isBetaTester", (NYT.column_name) = "is_beta_tester"];
    bool HasMusicSubscription = 9 [json_name = "hasMusicSubscription", (NYT.column_name) = "has_music_subscription"];
    uint64 MusicSubscriptionRegionId = 12 [json_name = "musicSubscriptionRegionId", (NYT.column_name) = "music_subscription_region_id"];
};
